Content.Client/UserInterface/Controls/FancyTree/FancyTree.xaml.cs

```diff
@@ -0,0 +1,283 @@
+/// </summary>
+[GenerateTypedNameReferences]
+public sealed partial class FancyTree : Control
+{
+    public const string StylePropertyLineWidth = "LineWidth";
+    public const string StylePropertyLineColor = "LineColor";
+    public const string StylePropertyIconColor = "IconColor";
+    public const string StylePropertyIconExpanded = "IconExpanded";
+    public const string StylePropertyIconCollapsed = "IconCollapsed";
+    public const string StylePropertyIconNoChildren = "IconNoChildren";
+
+    public readonly List<TreeItem> Items = new();
+
+    public event Action<TreeItem?>? OnSelectedItemChanged;
+
+    public int? SelectedIndex { get; private set; }
+
+    private bool _rowStyleUpdateQueued = true;
+
+    /// <summary>
+    ///     Whether or not to draw the lines connecting parents & children.
+    /// </summary>
+    public bool DrawLines = true;
+
+    /// <summary>
+    ///     Colour of the lines connecting parents & their child entries.
+    /// </summary>
+    public Color LineColor = Color.White;
+
+    /// <summary>
+    ///     Color used to modulate the icon textures.
+    /// </summary>
+    public Color IconColor = Color.White;
+
+    /// <summary>
+    ///     Width of the lines connecting parents & their child entries.
+    /// </summary>
+    public int LineWidth = 2;
+
+    // If people ever want to customize this, this should be a style parameter/
+    public const int Indentation = 16;
+
+    public const string DefaultIconExpanded = "/Textures/Interface/Nano/inverted_triangle.svg.png";
+    public const string DefaultIconCollapsed = "/Textures/Interface/Nano/triangle_right.png";
+    public const string DefaultIconNoChildren = "/Textures/Interface/Nano/triangle_right_hollow.svg.png";
+
+    public Texture? IconExpanded;
+    public Texture? IconCollapsed;
+    public Texture? IconNoChildren;
+
+    /// <summary>
+    ///     If true, tree entries will hide their icon if the texture is set to null. If the icon is hidden then the
+    ///     text of that entry will no longer be aligned with sibling entries that do have an icon.
+    /// </summary>
+    public bool HideEmptyIcon
+    {
+        get => _hideEmptyIcon;
+        set => SetHideEmptyIcon(value);
+    }
+    private bool _hideEmptyIcon;
+
+    public TreeItem? SelectedItem => SelectedIndex == null ? null : Items[SelectedIndex.Value];
+
+    /// <summary>
+    ///     If true, a collapsed item will automatically expand when first selected. If false, it has to be manually expanded by
+    ///     clicking on it a second time.
+    /// </summary>
+    public bool AutoExpand = true;
+
+    public FancyTree()
+    {
+        RobustXamlLoader.Load(this);
+        LoadIcons();
+    }
+
+    private void LoadIcons()
+    {
+        IconColor = TryGetStyleProperty(StylePropertyIconColor, out Color color) ? color : Color.White;
+        var res = IoCManager.Resolve<IResourceCache>();
```

Comment:
I know we're trying to remove static resolves but not sure if the preferred way is passing it in via ctor.

Reply:
Ah I should just be using `IoCManager.InjectDependencies(this);` anyways

Reply:
Also gonna change the style properties so that they just directly set the texture